Correct announcement of move when variations selected

ukscid opened the pull request:
Move was announced when the variation selection dialog popup. Now announce move after the selection with announcing the selected move.
Move was announced twice when the variation selection dialog was not used.

Commit:
Correct announcement of move when variations selected

Move was announced when the variation selection dialog popup. Now announce move after the selection.
Move was announced twice when the variation selection dialog was not used

benini committed:
Something like this should work

benini commented on tcl/main.tcl:
```diff
@@ -545,6 +545,7 @@ proc showVars {} {
         } else  {
             sc_var moveInto [expr $cur - 1]
         }
+        ::utils::sound::AnnounceForward [sc_game info previous]
```
It would be better to invoke ::move::Follow instead of manually entering into the var, announcing the move and updating the board.

ukscid replied:
Do you mean to rewrite `bind .variations <Right>` and use therefor  `::move::Follow` ?
move::Follow needs a uci move as input. Where should this come from?

benini replied:
There are multiple ways to enter into a variation (entering the move with the mouse or the keyboard, the varPopup, an external board, etc...) andi t is desireable to have the same behavior for all of them and use the same code.

I was thinking about replacing
$w.lbVar insert {} end -id $j ...
with
$w.lbVar insert {} end -id $move ...
and the just
::move::Follow [.variations.lbVar selection]

However it probably does not work if there are variations that start with the same move.

benini commented:
I have added a commit on how I think it should work.
Can you please try it and eventually fix it?

Commit:
small code corrections

ukscid commented:
I checked and tested your code and made small adjustment. This works.

benini committed:
Update move.tcl

I think expr should already return a bool and the ternary operator is not necessary

benini committed:
Update move.tcl

Final clean up, to be tested.

benini committed:
Update main.tcl

I do not want to spread announceMove calls.
I'll create a ::move::EnterVar function.
